### PR TITLE
PD-858: Add darkMode prop to Checkbox

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Refactors Checkbox component
- Adds darkMode prop to Checkbox

🎟 _Jira ticket:_ [PD-858](https://jira.mongodb.org/browse/PD-858)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/checkbox/src/Checkbox.tsx [hunk 1]
```diff
@@ -1,8 +1,7 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { createDataProp, IdAllocator } from '@leafygreen-ui/lib';
 import { css, cx } from '@leafygreen-ui/emotion';
-import { colors } from '@leafygreen-ui/theme';
```
Comment (lazytype) [inline packages/checkbox/src/Checkbox.tsx:5]: curious about the motivation for this change

File: packages/checkbox/package.json [hunk 2]
```diff
@@ -12,8 +12,10 @@
     "access": "public"
   },
   "dependencies": {
-    "@leafygreen-ui/lib": "^5.1.1",
-    "@leafygreen-ui/theme": "^2.0.1"
+    "@leafygreen-ui/lib": "^5.1.1"
+  },
+  "devDependencies": {
+    "@leafygreen-ui/palette": "^2.0.2"
```
Comment (DesignerDave) [inline packages/checkbox/package.json:18]: Why is this change being made?

Comment (brookescarlett) [reply packages/checkbox/package.json:18]: Isn't deprecating theme all together a goal? This is the last component that was relying on theme, so once we remove it here then we're done with the package throughout our repo. Further, the theme was only used in three places so I just replaced the reference to theme with the same hex value, with the hope that we'd move over to the palette entirely with this component eventually.

File: packages/checkbox/src/Checkbox.tsx (hunk 1 above)
Comment (brookescarlett) [reply packages/checkbox/src/Checkbox.tsx:5]: see below :)

File: packages/checkbox/package.json (hunk 2 above)
Comment (DesignerDave) [reply packages/checkbox/package.json:18]: Whoops, thought this was moving theme from a dependency to a devDependency. Why is this a devDependency I suppose is a better question 😅

Comment (brookescarlett) [reply packages/checkbox/package.json:18]: It's only being used in Storybook, for the three styles I changed, I inlined the hex values (so they're staying the same as they are)

**Review by lazytype — APPROVED** [review]

**Pull request merged**